src/args/types.mth

```diff
@@ -0,0 +1,107 @@
+module(args.types)
+
+import(prelude)
+import(data.maybe)
+import(data.str)
+
+import(args.parse)
+
+table(ArgumentParser)
+field(ArgumentParser~options, ArgumentParser, Maybe(List(ArgpOption)))
+field(ArgumentParser~parser, ArgumentParser, Maybe([ +ArgumentParser a Maybe(Str) ArgpOptionType -- +ArgumentParser a ]))
+field(ArgumentParser~args-doc, ArgumentParser, Maybe(Str))
+field(ArgumentParser~doc, ArgumentParser, Str)
+def(ArgumentParser.new, Maybe(List(ArgpOption)) Maybe([ +ArgumentParser a Maybe(Str) ArgpOptionType -- +ArgumentParser a ]) Maybe(Str) Str -- ArgumentParser,
+  ArgumentParser.alloc!
+  dup dip(~doc !)
+  dup dip(~args-doc !)
+  dup dip(~parser !)
+  dup dip(~options !)
+)
+
+def(ArgumentParser.parser, ArgumentParser -- Maybe([ +ArgumentParser a Maybe(Str) ArgpOptionType -- +ArgumentParser a ]), ~parser @)
+
+table(ArgumentParserState(a))
```

Comment:
The type parameter `(a)` here should not be allowed.
It's just ignored by the compiler, isn't it? Better to remove it for now. (& opened issue #196)
```suggestion
table(ArgumentParserState)
```